packages/core/workers/src/cpuCount.js

```diff
@@ -1,11 +1,71 @@
-const os = require('os');
+// @flow
+import os from 'os';
+import {execSync} from 'child_process';
 
-module.exports = function() {
-  let cores;
+const exec = (command: string): string => {
   try {
-    cores = require('physical-cpu-count');
-  } catch (err) {
-    cores = os.cpus().length;
+    let stdout = execSync(command, {
+      encoding: 'utf8',
+      // This prevents the command from outputting to the console
+      stdio: [null, null, null],
+    });
+    return stdout.trim();
+  } catch (e) {
+    return '';
   }
-  return cores || 1;
 };
+
+export function detectRealCores(): number {
+  let platform = os.platform();
+  let amount = 0;
+
+  if (platform === 'linux') {
+    let output =
+      exec('nproc') ||
```

Comment:
`nproc` returns the number of logical, not physical cores.

Reply:
It returned 6 for me when trying it out in alpine which is the amount of physical cores in my Macbook so I thought it logged the physical cores. Just double checked my mac is HyperThreaded so not entirely sure this is totally true.

Maybe my VM can't access all 12 virtual cores idk.

Would be great to know more about this.

Reply:
Dual-Core MBP:
```
$ nproc
4
$ sysctl -n hw.physicalcpu_max
2
```

Reply:
Guess it's a VM and Docker thing will update it

Reply:
Well, on Ubuntu (not a VM), it does return the number of physical cores (and this is the linux codepath).
I guess the manpage term `processing units` is interpreted differently across platforms.

Reply:
So maybe use nproc as fallback?